lib/rom/factory/attributes/association.rb

```diff
@@ -5,10 +5,9 @@ module Attributes
     # @api private
     module Association
       class << self
-        def new(assoc, builder, *args)
-          const_get(assoc.definition.type).new(assoc, builder, *args)
+        def new(assoc, ...)
+          const_get(assoc.definition.type).new(assoc, ...)
```

Comment:
Yes, I ran all the tests locally in 2.7 while working on this. I'm not aware of any behavioral changes to `...` between 2.7 and 3.0. Here's [the relevant changes entry](https://rubyreferences.github.io/rubychanges/2.7.html#keyword-argument-related-changes)

The big thing that changed behavior was inline pattern-matching; a failure to match throws an exception in 2.7 whereas it is a boolean in >= 3.0.

Reply:
Maybe it was changed before the release, I used to regularly check ruby's bugtracker back then. Thanks for checking

Reply:
Okay I found [this blogpost](https://blog.saeloun.com/2020/09/16/ruby-3.0-to-add-support-for-forwarding-arguments-along-with-lead-arguments.html) that mentions leading arguments before `...` may not work in 2.7.

I am doing that in at least one place so this might be a problem. It didn't fail any tests though, so I may need to add one.

Reply:
Alright I understand specifically what changed and when, which explains why I didn't encounter an error.

Leading arguments before `...` such as `new(assoc, ...)` was a syntax error originally. They added support later and backported to 2.7.2. I was running 2.7.6 so never noticed this.

If you run this with 2.7.0, however, it's still a syntax error. So I removed `...` and just used splats.

Good catch! 🙏🏽

Reply:
OMG, thank you for looking into it :)